drivers/led/is31fl319x.c

```diff
@@ -161,11 +152,32 @@ static int is31fl319x_set_color(const struct device *dev, uint32_t led, uint8_t
 	return ret;
 }
 
+static int is31fl319x_set_color(const struct device *dev, uint32_t led, uint8_t num_colors,
+				const uint8_t *color)
+{
+	const struct is31fl319x_config *config = dev->config;
+	const struct led_info *info = is31fl319x_led_to_info(config, led);
+	uint8_t channel_start;
+
+	if (info == NULL) {
+		return -ENODEV;
+	}
+
+	channel_start = is31fl319x_map_led_to_start_channel(config, led);
```

Comment:
I think there is a confusion here. The ->set_color() API method allows to configure the colors of a LED (for a given index). It should not overflow on the next LED. You can't handle LEDs as channels here.

Reply:
OK, understood.

Please use the following test instead:

```
        if (num_colors != info->num_colors) {
                LOG_ERR("%s: invalid number of colors: got=%d, expected=%d",
                        dev->name,
                        num_colors,
                        info->num_colors);
                return -EINVAL;
        }
```

The user must configure exactly the same number of colors that supported by the LED. If not, then `-EINVAL` is the correct error code to return.

In addition, please move this check before the call to `is31fl319x_map_led_to_start_channel().`